### PR TITLE
update(docs): update change log to Universum 0.19.0

### PR DESCRIPTION
**Pull request by k-dovgan**
Please submit **after** #465

### CODE REVIEW

**Comment by k-dovgan** [comment]
https://universum.readthedocs.io/en/update-changelog/changelog_ref.html

File: CHANGELOG.rst [hunk 1]
```diff
@@ -1,6 +1,39 @@
 Change log
 ==========
 
+0.19.0 (2020-08-??)
+-------------------
+
+BREAKING CHANGES
+~~~~~~~~~~~~~~~~
+
+* **migrate to Python3.7**
```
Comment (i-keliukh) [inline CHANGELOG.rst:10]: All of these need more detailed explanation.

File: CHANGELOG.rst [hunk 2]
```diff
@@ -1,6 +1,39 @@
...
+* **disable entry points and switch to ``python -m`` calls**
+* **rename ``_universum`` to ``universum``** *(configs need to be fixed)*
+
+New features
+~~~~~~~~~~~~
+
+* **github-handler:** add :doc:`a new Universum mode <args_github_handler>` to serve as a GitHub App
```
Comment (i-keliukh) [inline CHANGELOG.rst:17]: Why user need to know about this?
Also, add link to GitHub App docs.

File: CHANGELOG.rst [hunk 3]
```diff
@@ -1,6 +1,39 @@
...
+* **pylint:** allow selecting exact python version
```
Comment (i-keliukh) [inline CHANGELOG.rst:18]: AFAIR it allows selecting any version.
Previously it only allowed to select python 2 or 3.
I understand what is meant by "exact version", but it is not clear enough for reader.

File: CHANGELOG.rst [hunk 4]
```diff
@@ -1,6 +1,39 @@
...
+* **nonci:** set project root to current directory
```
Comment (i-keliukh) [inline CHANGELOG.rst:19]: Even I don't remember what is it.
Please provide more details.

File: CHANGELOG.rst [hunk 5]
```diff
@@ -1,6 +1,39 @@
...
+* **github_vcs:** aquire GitHub token only when needed
```
Comment (i-keliukh) [inline CHANGELOG.rst:20]: What is the benefit for user?

File: CHANGELOG.rst [hunk 6]
```diff
@@ -1,6 +1,39 @@
...
+
+Bug fixes
+~~~~~~~~~
+
+* **swarm:** only update status for latest revisions
```
Comment (i-keliukh) [inline CHANGELOG.rst:25]: Not clear enough.
What is the bug?

File: CHANGELOG.rst [hunk 7]
```diff
@@ -1,6 +1,39 @@
...
+* **nonci:** remove code report from nonci, launching steps twice
```
Comment (i-keliukh) [inline CHANGELOG.rst:26]: Please make it a proper sentence, for example:
fixed the issue with launching code report steps twice in nonci mode.

File: CHANGELOG.rst [hunk 8]
```diff
@@ -1,6 +1,39 @@
...
+* **report:** error message with successful build if not set to be reported
```
Comment (i-keliukh) [inline CHANGELOG.rst:27]: Please fix the grammar.
Also, what was the bug?

File: CHANGELOG.rst [hunk 9]
```diff
@@ -1,6 +1,39 @@
...
+* **setup:** make installing modules for GitHub VCS type non-mandatory
+* **docs:** limit nav panel depth to remove confusing subparagraphs
```
Comment (i-keliukh) [inline CHANGELOG.rst:29]: In my opinion this is an internal or too minor fix.
Also, it is not in the product itself.

File: CHANGELOG.rst [hunk 10]
```diff
@@ -1,6 +1,39 @@
...
+* **p4:** add disconnect before any connect to fix `connection lost` issues
+* **code_report:** replace code report pseudo-variable not only in ``command``
```
Comment (i-keliukh) [inline CHANGELOG.rst:31]: Perhaps add a name of the variable?
It would make the note more clear.

File: CHANGELOG.rst [hunk 11]
```diff
@@ -7,29 +7,35 @@ Change log
...
+    :class:`~universum.configuration_support.Variations` imports in configs need to be fixed
 
 New features
 ~~~~~~~~~~~~
 
-* **github-handler:** add :doc:`a new Universum mode <args_github_handler>` to serve as a GitHub App
-* **pylint:** allow selecting exact python version
+* **github-handler:** add :doc:`a new Universum mode <github_handler>`
+    to serve as a `GitHub App <https://docs.github.com/en/developers/apps>`__. GitHub Applications,
+    for example, are needed to run checks on newly commmited changes, or on changes to be merged yet
```
Comment (i-keliukh) [inline CHANGELOG.rst:22]: I think you don't need to add "for example" here.
As far as I understand, GitHub App is needed to create check suite (auto) and check runs and report their status.
We were able to run checks before we implemented GitHub App.

Comment (i-keliukh) [reply CHANGELOG.rst:22]: For readers I would add that github-handler and other changes in this version allow to implement full workflow for github.

File: CHANGELOG.rst [hunk 12]
```diff
@@ -7,29 +7,35 @@ Change log
...
 Bug fixes
 ~~~~~~~~~
 
-* **swarm:** only update status for latest revisions
-* **nonci:** remove code report from nonci, launching steps twice
-* **report:** error message with successful build if not set to be reported
+* **swarm:** only update status for latest revisions (as Swarm no longer supports outdated review status update)
+* **github_vcs:** acquire GitHub token when needed (so it no longer expires for long builds)
+* **nonci:** fix the issue with launching ``code_report=True`` steps twice
+* **report:** successful build not set to be reported no longer causes errors
```
Comment (i-keliukh) [inline CHANGELOG.rst:34]: errors in the terminal?

File: CHANGELOG.rst [hunk 13]
```diff
@@ -7,29 +7,35 @@ Change log
 BREAKING CHANGES
 ~~~~~~~~~~~~~~~~
 
-* **migrate to Python3.7**
-* **disable entry points and switch to ``python -m`` calls**
-* **rename ``_universum`` to ``universum``** *(configs need to be fixed)*
+* **migrate to Python3.7** due to Python2 is no longer supported
```
Comment (i-keliukh) [inline CHANGELOG.rst:10]: Please add space between Python and version number.

File: CHANGELOG.rst [hunk 14]
```diff
@@ -7,29 +7,35 @@ Change log
...
+    Python3.7 is now :doc:`required <prerequisites>` for Universum to work
+* **disable entry points** and switch to ``python3.7 -m universum`` calls
+    to avoid possible collisions, especially when using :doc:`analyzers <code_report>`
```
Comment (i-keliukh) [inline CHANGELOG.rst:13]: Not clear.
What collisions?
Also, users probably don't know what is "entry point" in this context.
Please add migration notice.

File: CHANGELOG.rst [hunk 15]
```diff
@@ -7,29 +7,35 @@ Change log
...
+* **rename `_universum` to `universum`**
+    :class:`~universum.configuration_support.Variations` imports in configs need to be fixed
```
Comment (i-keliukh) [inline CHANGELOG.rst:15]: Please make the migration notice more explicit.
I think it is better to just give direct order, like this:
> change the following line <...> to this <...>

File: CHANGELOG.rst [hunk 16]
```diff
@@ -7,31 +7,36 @@ Change log
...
+   ``from _universum.configuration_support import Variations``
 
 New features
 ~~~~~~~~~~~~
 
 * **github-handler:** add :doc:`a new Universum mode <github_handler>`
-    to serve as a `GitHub App <https://docs.github.com/en/developers/apps>`__. GitHub Applications,
-    for example, are needed to run checks on newly commmited changes, or on changes to be merged yet
+   to serve as a `GitHub App <https://docs.github.com/en/developers/apps>`__. GitHub Applications
+   are needed to run checks and report their status to GitHub for changes newly commmited or those to be merged yet.
```
Comment (i-keliukh) [inline CHANGELOG.rst:26]: "for changes newly commmited or those to be merged yet" -> "for changes on review and for committed changes"

File: CHANGELOG.rst [hunk 17]
```diff
@@ -7,31 +7,36 @@ Change log
...
+   With this mode it is possible to implement full GitHub workflow with Universum
 * **pylint:** allow selecting any Python version for checks
-    e.g. ``3.5`` instead of simple ``2``/``3`` switch earlier
+   e.g. ``3.5`` instead of simple ``2``/``3`` switch earlier
```
Comment (i-keliukh) [inline CHANGELOG.rst:29]: "earlier" -> "that was available before"

File: CHANGELOG.rst [hunk 18]
```diff
@@ -7,31 +7,36 @@ Change log
...
+   to simply run ``universum nonci`` from sources location without :doc:`setting <args>` ``--project-root`` manually
 
 Bug fixes
 ~~~~~~~~~
 
 * **swarm:** only update status for latest revisions (as Swarm no longer supports outdated review status update)
 * **github_vcs:** acquire GitHub token when needed (so it no longer expires for long builds)
 * **nonci:** fix the issue with launching ``code_report=True`` steps twice
-* **report:** successful build not set to be reported no longer causes errors
+* **report:** successful build not set to be reported no longer causes errors in the terminal
```
Comment (i-keliukh) [inline CHANGELOG.rst:39]: Maybe "if the report is not enabled for successful builds, it doesn't cause errors in the terminal anymore"?

**Review by i-keliukh — APPROVED** [review]

**Pull request merged**